app/domain_models/follow_status_data.rb

```diff
@@ -0,0 +1,83 @@
+class FollowStatusData
+
+    attr_reader :current_status,:current_following,:index
+  
+    def initialize()#將資料庫的紀錄轉換成當下的追蹤狀態
+        @current_status={} #每個使用者的追蹤清單
+        @current_following={}    #每個使用者的粉絲清單
+        @index=1
+        user_number=User.count
+
+        1.upto(user_number) do |user_id|
```

Comment:
Why you use loop here?
We barely use the method, upto.
Maybe it is not so necessary to use loop here.

Reply:
The purpose of the loop is to assign an empty array as the value to every user_id(key). In the next block, multiple integers will be assigned to these arrays, representing the following and follower lists for each users. I tried to use other function to replace the loop but it turned out to be weirdly buggy, so I guess the loop is necessary.